flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/DirectConvertRule.java

```diff
@@ -160,6 +160,9 @@
         DEFINITION_OPERATOR_MAP.put(BuiltInFunctionDefinitions.HEX, FlinkSqlOperatorTable.HEX);
         DEFINITION_OPERATOR_MAP.put(
                 BuiltInFunctionDefinitions.TRUNCATE, FlinkSqlOperatorTable.TRUNCATE);
+        DEFINITION_OPERATOR_MAP.put(
```

Comment:
Implementing the functions here and in `FlinkSqlOperatorTable` duplicates their type strategy information from the new function stack. We can avoid that by not adding this in those two places, however it requires https://github.com/apache/flink/pull/16046 (something I discussed with @twalthr). It would be nice, though, to not duplicate the type signatures for these functions and implementing them in both the old and new stack.

Ideally we'd even implement GREATEST and LEAST entirely on the new stack, but I think this isn't yet possible, so going through code generation should be fine for now.

Reply:
Let's wait #16046 merged first